File: svm/javascript/svm.js
class SVM {

    constructor(C, kernel) {
      this.C = C;
      this.kernel = kernel;
      this.w = [0, 0];
      this.b = 0;
    }
  
    fit(X, y, n) {
      for (let i = 0; i < n; i++) {
        let score = 0;
        for (let j = 0; j < 2; j++) {
          score += this.w[j] * X[i][j];
        }
        if (y[i] * score + this.b <= 1) {
          for (let j = 0; j < 2; j++) {
            this.w[j] += this.C * y[i] * X[i][j];
          }
          this.b += this.C * y[i];
        }
      }
    }
  
    predict(X) {
      let score = 0;
      for (let j = 0; j < 2; j++) {
        score += this.w[j] * X[j];
      }
      return score >= 0 ? 1 : -1;
    }
  
  }
  
  const svm = new SVM(1.0, "linear");
  svm.fit([[1, 2], [3, 4]], [1, -1], 2);
  console.log(svm.predict([5, 6])); // 1
  

File: svm/julia/svm.jl
struct SVM
    C::Float64
    kernel::String
    w::Vector{Float64}
    b::Float64
end

function SVM(C::Float64, kernel::String)
    SVM(C, kernel, zeros(Float64, 2), 0.0)
end

function fit(svm::SVM, X::Array{Float64}, y::Array{Int}, n::Int)
    for i in 1:n
        score = 0.0
        for j in 1:2
            score += svm.w[j] * X[i, j]
        end
        if y[i] * score + svm.b <= 1.0
            for j in 1:2
                svm.w[j] += svm.C * y[i] * X[i, j]
            end
            svm.b += svm.C * y[i]
        end
    end
    svm
end

function predict(svm::SVM, X::Array{Float64})
    score = 0.0
    for j in 1:2
        score += svm.w[j] * X[j]
    end
    score >= 0 ? 1 : -1
end

svm = SVM(1.0, "linear")
svm = fit(svm, [[1, 2], [3, 4]], [1, -1], 2)
println(predict(svm, [5, 6])) # prints "1"
